fennel/datasets/test_schema_validator.py

```diff
@@ -1061,3 +1061,204 @@ def create_dataset(cls, activity: Dataset):
         str(e.value)
         == "Field `userid` is a key or timestamp field in schema of assign node input '[Dataset:RatingActivity]'. Value fields are: ['rating', 'movie']"
     )
+
+
+def test_window_without_key_fails():
+    with pytest.raises(ValueError) as e:
+
+        @meta(owner="nitin@fennel.ai")
+        @dataset
+        class PageViewEvent:
+            user_id: str
+            page_id: str
+            t: datetime
+
+        @meta(owner="nitin@fennel.ai")
+        @dataset
+        class Sessions:
+            user_id: str
+            window: WindowStruct
+            t: datetime
+
+            @pipeline(version=1)
+            @inputs(PageViewEvent)
+            def pipeline_window(cls, app_event: Dataset):
+                return app_event.groupby().window(type="session", gap="10m")
+
+    assert str(e.value) == """Must specify at least one key"""
```

Comment:
This is error contains very little context, also considering moving invalid tests that are not schema related to test_invalid_datasets

Reply:
Can you share an example with of window operator with respect to invalid_dataset?